scripts/ci/build.sh

```diff
@@ -2,17 +2,24 @@
 set -ex
 
 ROOTDIR="$(dirname "$0")/../.."
+PRE_RELEASE="$1"
+
 cd "${ROOTDIR}"
 
 # shellcheck disable=SC2166
 if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]
 then
     echo -n "" >prerelease.txt
 else
-    # Use last commit date rather than build date to avoid ending up with builds for
-    # different platforms having different version strings (and therefore producing different bytecode)
-    # if the CI is triggered just before midnight.
-    TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
+    # Use the given prerelease string if provided
+    if [[ "$PRE_RELEASE" =~ ^[a-zA-Z]+\.[0-9]{4}\.[0-9]{1,2}\.[0-9]{1,2}$ ]]; then
+        echo "$PRE_RELEASE" >prerelease.txt
+    else
+        # Use last commit date rather than build date to avoid ending up with builds for
+        # different platforms having different version strings (and therefore producing different bytecode)
+        # if the CI is triggered just before midnight.
+        TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
```

Comment:
I think it would be better if the parameter allowed replacing only the `ci` part here, not the whole pre-release string. This way the caller would not have to worry about getting the date right. It's just too easy to get the date wrong (note for example that it's the date of the commit, not the current date).

Reply:
Yes, it is much better indeed. I will do the change.